crates/indexer/src/database.rs

```diff
@@ -60,19 +68,53 @@ pub fn execute_with_better_error<
 >(
     conn: &mut PgConnection,
     query: diesel::query_builder::InsertStatement<T, U>,
+    mut additional_where_clause: Option<&'static str>,
 ) -> diesel::QueryResult<usize>
 where
     <T as diesel::QuerySource>::FromClause: diesel::query_builder::QueryFragment<diesel::pg::Pg>,
 {
-    let debug = diesel::debug_query::<diesel::pg::Pg, _>(&query).to_string();
+    let original_query = diesel::debug_query::<diesel::pg::Pg, _>(&query).to_string();
+    // This is needed because if we don't insert any row, then diesel makes a call like this
+    // SELECT 1 FROM TABLE WHERE 1=0
+    if original_query.to_lowercase().contains("where") {
+        additional_where_clause = None;
+    }
+    let final_query = match additional_where_clause {
+        Some(where_clause) => UpsertFilterLatestTransactionQuery {
+            query,
+            where_clause,
+        },
+        None => UpsertFilterLatestTransactionQuery {
+            query,
+            where_clause: "",
+        },
+    };
+    let debug = diesel::debug_query::<diesel::pg::Pg, _>(&final_query).to_string();
     aptos_logger::debug!("Executing query: {:?}", debug);
```

Comment:
I guess this only prints out when debug is enabled?

You can also put this directly in the `aptos_logger::debug!`, and it's supposed to not even process it if the log is disabled.

Reply:
good point.